packages/neon_framework/packages/neon_http_client/lib/src/interceptors/authentication_interceptor.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:async';
+
+import 'package:http/http.dart';
+import 'package:meta/meta.dart';
+import 'package:neon_http_client/src/interceptors/http_interceptor.dart';
+import 'package:nextcloud/nextcloud.dart';
+
+@internal
+final class AuthenticationInterceptor implements HttpInterceptor {
+  var _blocked = false;
+
+  @override
+  bool shouldInterceptRequest(BaseRequest request) {
+    return (request.headers['authorization'] ?? '').isNotEmpty && _blocked;
+  }
```

Comment:
```suggestion
  bool shouldInterceptRequest(BaseRequest request) {
  	final authorization = request.headers['authorization'];
    return authorization != null  && authorization.isNotEmpty && _blocked;
  }
```

You can also use pattern matching if you prefer to.